### ISSUE TITLE
FSharp nupkgs showing up as prebuilts on .NET 8 RC2 source-build artifacts

### ISSUE THREAD

**Issue by mateusrodrigues** [dotnet/source-build#3689]
### Describe the Bug

An offline build of the VMR on tag `v8.0.0-rc.2.23479.6` outputs `Private.SourceBuilt.Prebuilts.8.0.100-rc.2.23502.1.ubuntu.23.10-x64.tar.gz`, which contains the following files:

```
$ tar tf Private.SourceBuilt.Prebuilts.8.0.100-rc.2.23502.1.ubuntu.23.10-x64.tar.gz 
./
./FSharp.Compiler.Service.43.8.100.nupkg
./FSharp.Core.8.0.100.nupkg
./FSharp.Core.8.0.100-beta.23475.2.nupkg
./FSharp.Compiler.Service.43.8.100-preview.23475.2.nupkg
```

I've run a quick checksum comparison between these files and the ones with the same name from the Artifacts tarball and they seem to be the same.

### Steps to Reproduce

This was a stage 2 kind of build, built against the currently published .NET 8 RC2 package on the [Ubuntu archive](https://launchpad.net/ubuntu/+source/dotnet8/8.0.100-8.0.0~rc2-0ubuntu1).

**Step 1**: installed dependencies
```
$ sudo apt install clang cmake bash-completion debhelper-compat dotnet-sdk-8.0 dotnet-sdk-8.0-source-built-artifacts git libicu-dev libkrb5-dev liblttng-ust-dev libssl-dev libunwind-dev lld llvm locales-all python3 zlib1g-dev
```

**Step 2**: linked artifacts tarball
```
$ ln -s /usr/lib/dotnet/source-built-artifacts/Private.SourceBuilt.Artifacts.8.0.100-rc.2.23502.1.ubuntu.23.10-x64.tar.gz prereqs/packages/archive/
```

**Step 3**: copied SDK file tree to current folder
```
$ cp -RLp /usr/lib/dotnet/ previously-built-dotnet
```

**Step 4**: ran build script
```
$ VERBOSE=1 ./build.sh --with-sdk previously-built-dotnet/ -- /v:n         /p:SkipPortableRuntimeBuild=true        /p:LogVerbosity=n       /p:MinimalConsoleLogOutput=false       /p:ContinueOnPrebuiltBaselineError=true
```

### Other Information

Seems to be related to #2989

**Comment by MichaelSimons** [comment]
Yes, this is the same issue as https://github.com/orgs/dotnet/projects/101/views/1?filterQuery=fsharp&pane=issue&itemId=8606342.  These are not actual prebuilts.  This was fixed in main (.NET 9.0).   This issue has existed for a long time now - it affects 6.0 and 7.0 as well.  @mateusrodrigues how impactful is this for you?   Trying to gauge if this fix needs to be backported or not.

**Comment by mateusrodrigues** [comment]
Not impactful at all. We are simply ignoring the Prebuilts tarball and only shipping the Artifacts one. I guess we can close this one since it has already been fixed in main. Thanks!
